packages/xstate-test/src/machine.ts

```diff
@@ -0,0 +1,135 @@
+import { serializeState, SimpleBehavior } from '@xstate/graph';
+import {
+  ActionObject,
+  AnyEventObject,
+  AnyState,
+  AnyStateMachine,
+  createMachine,
+  EventFrom,
+  EventObject,
+  StateFrom,
+  TypegenConstraint,
+  TypegenDisabled
+} from 'xstate';
+import { TestModel } from './TestModel';
+import {
+  TestMachineConfig,
+  TestMachineOptions,
+  TestModelEventConfig,
+  TestModelOptions
+} from './types';
+import { flatten } from './utils';
+import { validateMachine } from './validateMachine';
+
+export async function testStateFromMeta(state: AnyState) {
+  for (const id of Object.keys(state.meta)) {
+    const stateNodeMeta = state.meta[id];
+    if (typeof stateNodeMeta.test === 'function' && !stateNodeMeta.skip) {
+      await stateNodeMeta.test(state);
+    }
+  }
+}
+
+export function createTestMachine<
```

Comment:
what is the exact purpose of this - I mean: at the moment both `createMachine` and `createTestMachine` will be accepted as arguments for `createTestModel`, right? What is going to be put in the docs etc about this?

Reply:
`createTestMachine` provides additional constraints on the machines that `createTestModel(...)` accepts, to prevent footguns

Reply:
if there are footguns with `createMachine` - shouldn't we disallow using `createMachine` together with `createTestModel`?

Reply:
I don't think so, for migration reasons. I think we do want to allow cases where you create the machine in XState and pass it to `createTestModel` - even if there are footguns.